app/models/transformation_mapping.rb

```diff
@@ -6,4 +6,72 @@ class TransformationMapping < ApplicationRecord
   def destination(source)
     transformation_mapping_items.find_by(:source => source).try(:destination)
   end
+
+  # vm_list: collection of hashes, each descriping a VM.
+  def validate_vms(vm_list = nil)
+    vm_list.present? ? identify_vms(vm_list) : select_vms
+  end
+
+  private
+
+  def select_vms
+    # TBD
+  end
+
+  def identify_vms(vm_list)
+    valid_list = []
+    invalid_list = []
+    conflict_list = []
+
+    vm_list.each do |row|
+      vm_name = row['Name']
+
+      query = Vm.where(:name => vm_name)
+      query = query.where(:uid_ems => row['UID']) if row['UID'].present?
+      query = query.where(:host => {:name => row['Host']}) if row['Host'].present?
+      query = query.where(:ext_management_system => {:name => row['Provider']}) if row['Provider'].present?
```

Comment:
I don't like that the incoming vm_list array of hashes has "human readable" strings (aka presentation strings).  These should ideally be lower-cased strings or symbols and probably should align tightly to the column names.  In fact, if the incoming vms_list was a hash where the column names were keys, that could be passed *directly* to the query.  In other words, the 4 lines above are really a conversion from "csv -formatted" to "database-formatted", which IMO doesn't belong here...the csv -> hash conversion belongs with something like a class that deals with the CSV details itself.